app/controllers/results_controller.rb

```diff
@@ -498,7 +498,7 @@ def download_zip
         revision = repo.get_revision(revision_identifier)
         repo.send_tree_to_zip(assignment.repository_folder, zip_file, zip_name, revision) do |file|
           submission_file = files.find_by(filename: file.name, path: file.path)
-          submission_file.retrieve_file(params[:include_annotations] == 'true' && !submission_file.is_supported_image?)
+          submission_file&.retrieve_file(params[:include_annotations] == 'true' && !submission_file.is_supported_image?)
```

Comment:
Do we need to modify `send_tree_to_zip` to handle a `nil` value? You've probably tested this, I just wanted to double-check.

Reply:
`nil` values are written as the empty string which in this case in the intended outcome